packages/graphql/src/translate/connection/create-connection-and-params.ts

```diff
@@ -89,143 +89,148 @@ function createConnectionAndParams({
         const unionSubqueries: string[] = [];
 
         unionNodes.forEach((n) => {
-            const relatedNodeVariable = `${nodeVariable}_${n.name}`;
-            const nodeOutStr = `(${relatedNodeVariable}:${n.name})`;
+            if (!whereInput || Object.prototype.hasOwnProperty.call(whereInput, n.name)) {
+                const relatedNodeVariable = `${nodeVariable}_${n.name}`;
+                const nodeOutStr = `(${relatedNodeVariable}:${n.name})`;
+
+                const unionSubquery: string[] = [];
+                const unionSubqueryElementsToCollect = [...elementsToCollect];
+
+                const nestedSubqueries: string[] = [];
+
+                if (node) {
+                    // Doing this for unions isn't necessary, but this would also work for interfaces if we decided to take that direction
```

Comment:
I don't understand this, if it's not necessary for unions why are we doing it? Could this comment be misleading.

Reply:
I actually think it is necessary, so comment removed... 🤫